cmd/checkertools/new.go

```diff
@@ -0,0 +1,126 @@
+package checkertools
+
+import (
+	"bytes"
+	"flag"
+	"go/build"
+	"go/format"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// TODO: add more such tools, like checker-rename to easily rename
+// checker type, update comments and rename testdata dir?
+
+// NewMain implements gocritic sub-command entry point.
+func NewMain() {
+	flag.Usage = func() {
+		log.Printf("usage: rulename")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+	// TODO: add camel case check.
+	ruleName := flag.Arg(0)
+
+	// We expect gocritic sources to be under canonical import
+	// path under GOPATH.
+	root := filepath.Join(build.Default.GOPATH,
+		"src", "github.com", "go-critic", "go-critic")
+
+	testDir := createTestDir(ruleName, root)
+	createPositiveTests(testDir)
+	createNegativeTests(testDir)
+	createCheckerFile(ruleName, root)
+}
+
+func writeFile(filename string, data []byte) error {
+	log.Printf("%s: new file", filename)
+	return ioutil.WriteFile(filename, data, 0666)
+}
+
+func makeDir(filename string) error {
+	log.Printf("%s: new dir", filename)
+	return os.Mkdir(filename, 0755)
+}
+
+func createTestDir(ruleName, root string) (testDir string) {
+	testDir = filepath.Join(root, "cmd", "gocritic", "testdata", ruleName)
+	if err := makeDir(testDir); err != nil {
+		log.Fatalf("create test dir: %v", err)
+	}
+	return testDir
+}
+
+func createPositiveTests(testDir string) {
+	const data = `package checker_test
+
+// TODO: write tests that trigger newly added checker.
+
+/// this is example of warning directive. It fails your tests right now
```

Comment:
What about note about `///` ? 😉

Reply:
Or explain what this line should contain, ex warning message.

Reply:
We should decide whether this is a tutorial or stub generator.
Or add `-with-help` flag that is true by default.